src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -297,6 +304,22 @@ namespace AppInstaller::CLI::Workflow
         context <<
             PortableInstallImpl <<
             ReportInstallerResult("Portable"sv, APPINSTALLER_CLI_ERROR_PORTABLE_INSTALL_FAILED, true);
+
+        const auto& installReturnCode = context.Get<Execution::Data::OperationReturnCode>();
+
+        if (installReturnCode != 0)

```

Comment:
I don't think that an `upgrade` should uninstall if it fails.  It should probably just leave things how they are for the user to sort out.

Reply:
Added a check that only performs a cleanup if install fails and the operation is not an upgrade.